fastlane_bot/modes/base_triangle.py

```diff
@@ -174,6 +174,141 @@ def get_combos(
                             combos,
                         )
         return combos
+    
+    def sort_pairs(self, pairs):
+        # Clean up the pairs alphabetically
+        return ["/".join(sorted(pair.split('/'))) for pair in pairs]
+
+    def flatten_nested_items_in_list(self, nested_list):
+        # unpack nested items
+        flattened_list = []
+        for items in nested_list:
+            flat_list = []
+            for item in items:
+                if isinstance(item, list):
+                    flat_list.extend(item)
+                else:
+                    flat_list.append(item)
+            flattened_list.append(flat_list)
+        return flattened_list
+    
+    def get_triangle_groups(self, flt, x_y_pairs):
+        # Get groups of triangles that conform to (flt/x , x/y, y/flt) where x!=y
+        triangle_groups = []
+        for pair in x_y_pairs:
+            x,y = pair.split('/')
+            triangle_groups += [("/".join(sorted([flt,x])), pair, "/".join(sorted([flt,y])))]
+        return triangle_groups
+
+    def get_all_relevant_pairs_info(self, CCm, all_relevant_pairs):
+        # Get pair info for the cohort to allow decision making at the triangle level
+        all_relevant_pairs_info = {}
+        for pair in all_relevant_pairs:            
+            all_relevant_pairs_info[pair] = {}
+            pair_curves = CCm.bypair(pair)
+            carbon_curves = []
+            non_carbon_curves = []
+            for x in pair_curves:
+                if x.params.exchange in self.ConfigObj.CARBON_V1_FORKS:
+                    carbon_curves += [x]
+                else:
+                    non_carbon_curves += [x]
+            all_relevant_pairs_info[pair]['non_carbon_curves'] = non_carbon_curves
+            all_relevant_pairs_info[pair]['carbon_curves'] = carbon_curves
+            all_relevant_pairs_info[pair]['curves'] = non_carbon_curves + [carbon_curves] if len(carbon_curves) >0 else non_carbon_curves  # condense carbon curves into a single list
```

Comment:
1. A space is missing in `>0`
2. I'm pretty sure that you should change `[carbon_curves]` to `carbon_curves`
3. It looks like you can get rid of the entire if/else here, and just leave the first part (left of the `if`)

In 3, I'm assuming that you've meant the `+` to be of higher precedence than the `if`.
So you might want to verify this, assuming that you did not intend to get something like:
```
non_carbon_curves + non_carbon_curves
```

Reply:
1. Fixed
2. No the carbon curves are condensed to a single list (hence the comment at the end of the line
3. The desired behavior is to populate the dictionary as follows:
     if there are no carbon_curves then simply put in non_carbon_curves
    if there are carbon_curves then condense to a single list ([carbon_curves]) and add it to the non_carbon_curves list
Happy to hear suggestion though I did verify the behavior